pmdarima/preprocessing/endog/tests/test_fourier.py

```diff
@@ -0,0 +1,45 @@
+# -*- coding: utf-8 -*-
+
+import numpy as np
+from numpy.testing import assert_array_almost_equal
+import pytest
+
+from pmdarima.preprocessing import FourierEndogTransformer
+
+
+@pytest.mark.parametrize(
+    'exog', [
+        None,
+        np.random.rand(5, 3),
+    ]
+)
+def test_invertible(exog):
+    y = np.arange(5)
+    trans = FourierEndogTransformer()
+    y_t, e_t = trans.fit_transform(y, exog)
+    y_prime, e_prime = trans.inverse_transform(y_t, e_t)
+
+    assert_array_almost_equal(y, y_prime)
+
+    # exog should all be the same too
+    if exog is None:
+        assert exog is e_t is e_prime is None
+    else:
+        assert_array_almost_equal(exog, e_t)
+        assert_array_almost_equal(exog, e_prime)
+
+
+def test_value_error_when_y_is_none():
+    y = None
+    exog = None
+    trans = FourierEndogTransformer()
+    with pytest.raises(ValueError):
+        trans.fit_transform(y, exog)
+
+
+def test_type_error_when_y_is_not_real():
+    y = [4. + 0.j, 8. + 12.j, 16. + 0.j, 8. - 12.j]
```

Comment:
Nice test!

Reply:
Directly from the [SciPy examples](https://docs.scipy.org/doc/scipy/reference/generated/scipy.fftpack.rfft.html#scipy.fftpack.rfft]) 😂